x/evm/keeper/msg_server.go

```diff
@@ -176,21 +177,23 @@ func (server msgServer) EVMTransaction(goCtx context.Context, msg *types.MsgEVMT
 			},
 		)
 
-	} else {
-		// if applyErr is nil then res must be non-nil
-		if res.Err != nil {
-			serverRes.VmError = res.Err.Error()
+		return
+	}
+	serverRes.GasUsed = res.UsedGas
+	serverRes.ReturnData = res.ReturnData
+	serverRes.Logs = types.NewLogsFromEth(stateDB.GetAllLogs())
```

Comment:
is this and the above assignments safe to call prior to the `res.Err` check given that its set afterwards in the prior impl?